modules/new_profile/php/new_profile.class.inc

```diff
@@ -44,19 +53,278 @@ class New_Profile extends \NDB_Page
+    {
+        $this->error = null;
+        $errors      = array();
+        $config      = \NDB_Config::singleton();
+        $user        = \User::singleton();
+        $db          = \Database::singleton();
+
+        // Validate DOB fields
+        if (empty($values['dobDate'])) {
+            $errors['dobDate'] = 'Date of Birth field is required';
+        }
+        if (empty($values['dobDateConfirm'])) {
+            $errors['dobDateConfirm'] = 'Confirm Date of Birth field is required';
+        }
+        if ($values['dobDate'] != $values['dobDateConfirm']) {
+            $errors['dobDate'] = 'Date of Birth fields must match.';
+        }
+
+        // Validate EDC fields
+        if ($config->getSetting('useEDC') == "true") {
+            if (empty($values['edcDate'])) {
+                $errors['edcDate'] = 'Expected Date of Confinement field is required';
+            }
+
+            if (empty($values['edcDateConfirm'])) {
+                $errors['edcDateConfirm'] = 'Confirm EDC field is required';
+            }
+
+            if ($values['edcDate'] != $values['edcDateConfirm']) {
+                $errors['edDate'] = 'Expected Date of Confinement fields must match.';
+            }
+        }
+
+        // Validate sex fields
+        if (empty($values['sex'])) {
+            $errors['sex'] = 'Sex is required.';
+        }
+
+        // Validate pscid
+        $PSCIDsettings = $config->getSetting('PSCID');
+        if ($PSCIDsettings['generation'] == 'user') {
+            if (empty($values['site'])) { // user is in only one site
+                $centerIDs = $user->getData('CenterIDs');
+                $centerID  = $centerIDs[0];
+                $site      =& \Site::singleton($centerID);
+            } else {
+                // user has multiple sites,
+                // so validate PSCID against the Site selected
+                $site =& \Site::singleton($values['site']);
+            }
+
+            if (empty($values['pscid'])) {
+                $errors['pscid'] = 'PSCID must be specified';
+            } elseif (!\Candidate::validatePSCID(
+                $values['pscid'],
+                $site->getSiteAlias()
+            )
+            ) {
+                $errors['pscid'] = 'PSCID does not match the required structure';
+            } elseif ($db->pselectOne(
+                "SELECT count(PSCID) FROM candidate WHERE PSCID=:V_PSCID",
+                array('V_PSCID' => $values['pscid'])
+            ) > 0
+            ) {
+                $errors['pscid'] = 'PSCID has already been registered';
+            }
+        }
+
+        // Validate site entered
+        if (empty($values['site'])) {
+            $errors['site'] = 'Site is required.';
+        } else {
+            $psc = (int)$values['site'];
+            $user_list_of_sites = $user->getData('CenterIDs');
+            $num_sites          = count($user_list_of_sites);
+            if ($num_sites > 1 && !$user->hasCenter($psc)) {
```

Comment:
> I think this should only be true if the user has 1 site. If it has more, it should be an error to not pass one.

@driusan yes, you're right. the validation is done here so i'll remove the case in `_process()`